libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -288,6 +295,22 @@ void AP_ICEngine::update(void)
         }
     }
 
+    float rpm_value;
+    if (redline_rpm > 0 && rpm.get_rpm(rpm_instance-1, rpm_value)) {
+        if (rpm_value > redline_rpm && !(options & uint16_t(Options::REDLINE_WARNING))) {
```

Comment:
as an RPM creeps up, it can spam the GCS. This needs a don't-spam-GCS-faster-than-1-sec type inhibitor. This can't just drop the GCS send attempt because you still want to know the final above/below state sent to the GCS.

Reply:
The other option I was thinking about was a deadband on the way down. Put in the parameter description that the deadband should be larger than the noise.

Reply:
nah, that seems unnecessary. If you do want to do that, perhaps make it a few percent of the redline. No need to make a param for it. Also, This is likely a noisy measurement so you may want to LPF it. Alarm on peak (or fast LPF) and de-alarm on slow LPF so you know it's truly out of the redline zone

Reply:
Done. 10% deadband and 1 pole 5Hz lowpass filter for the RPM value.